### ISSUE TITLE
Add automated golint on pull requests

### ISSUE THREAD

**Issue by crosbymichael** [docker/swarm#499]
**Problem**

Automated golint on pull requests

**Solution**

Add golint to travis.yml

**Comment by technolo-g** [comment]
#dibs

**Comment by gegere** [comment]
#dibs next...

**Comment by technolo-g** [comment]
PR submitted (https://github.com/docker/swarm/pull/514), but depends on Issue #500 before it can be merged. If there is no PR by @peggyl (no rush :) ) maybe we can knock that out together @gegere ?

**Comment by peggyl** [comment]
@technolo-g @gegere Sorry for the delayed response; a few other things came up this week and I didn't get a chance to revisit this after the Docker SF event on Wednesday. 

I started on #500 but so far have mostly fixed the naming/casing issues. The remaining golint fixes are for adding comments on exported functions and variables, which is a bit more non-trivial since it requires reading enough of the source code to include accurate/helpful comments. Still working on this, but I'll open a PR for what I have tonight and comment on #500 so others can work on it as well. Might be easier to split it up by package?

**Comment by gegere** [comment]
@technolo-g @peggyl I am still pretty new to Go and thought this would be something I could tackle at the hackathon. Any chance a future hangout would work and explain a few details. Happy to be a grunt, need to take on some more knowledge. Thanks!
